app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.java

```diff
@@ -170,24 +172,35 @@ public ViewModel(final @NonNull Environment environment) {
         .compose(bindToLifecycle())
         .subscribe(__ -> this.showStarredPrompt.onNext(null));
 
-      loggedOutUserOnStarClick
+      currentProject
+        .compose(combineLatestPair(this.currentConfig.observable().map(Config::countryCode)))
         .compose(bindToLifecycle())
-        .subscribe(__ -> this.showLoginTout.onNext(null));
-
-      this.shareButtonClickedSubject
+        .subscribe(this.projectAndUserCountry::onNext);
+
+      this.playVideo = currentProject.compose(takeWhen(this.playVideoButtonClicked));
+      this.showShareSheet = currentProject.compose(takeWhen(this.shareButtonClicked));
+      this.startCampaignWebViewActivity = currentProject.compose(takeWhen(this.blurbTextViewClicked));
+      this.startCheckoutActivity = currentProject.compose(takeWhen(this.backProjectButtonClicked));
+      this.startCreatorBioWebViewActivity = currentProject.compose(takeWhen(this.creatorNameTextViewClicked));
+      this.startCommentsActivity = currentProject.compose(takeWhen(this.commentsTextViewClicked));
+      this.startManagePledgeActivity = currentProject.compose(takeWhen(this.managePledgeButtonClicked));
+      this.startProjectUpdatesActivity = currentProject.compose(takeWhen(this.updatesTextViewClicked));
+      this.startViewPledgeActivity = currentProject.compose(takeWhen(this.viewPledgeButtonClicked));
```

Comment:
if you're wondering where all these "new" outputs are coming from they are not new we used to do all this logic within the getter functions with our `this.project` BehaviorSubject 😱